### PR TITLE
Add multiple parameter value classes

### PR DESCRIPTION
**Pull request by winder**
Created a number of ParameterValue types to make the contract implementation slightly less verbose.

### CODE REVIEW

File: src/main/java/com/algorand/algosdk/templates/ContractTemplate.java [hunk 1]
```diff
@@ -14,44 +14,66 @@
 
 public class ContractTemplate {
 
-	public enum DataType {
-		BASE64, ADDRESS, INT;
+	/**
+	 * Values are appended to a program at specific offsets. Depending on the type the source template has placeholders
```
Comment (algonautshant) [inline src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: Do you want to mention here something that these are not cross-SDK compatible? That each binary works only within the SDK it was designed for, and that the placeholder sizes are also SDK specific?

File: src/main/java/com/algorand/algosdk/templates/ContractTemplate.java [hunk 2]
```diff
@@ -14,44 +14,66 @@
...
+	 * of differing sizes, this offset is returned by the "updateProgram" method and is used to traverse the template.
+     *
+	 * Offsets are the following:
```
Comment (algonautshant) [inline src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: I do not think the use of the term "Offsets" here is correct. The following of the placeholder sizes, and not offsets.

File: src/main/java/com/algorand/algosdk/templates/ContractTemplate.java (hunk 1 above)
Comment (winder) [reply src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: Good point, clarification added.

File: src/main/java/com/algorand/algosdk/templates/ContractTemplate.java (hunk 2 above)
Comment (winder) [reply src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: Yes, you're right. I've tried to clarify this.

**Comment by winder** [comment]
@algonautshant thanks for the feedback, I've made changes to address the problems.

File: src/main/java/com/algorand/algosdk/templates/ContractTemplate.java [hunk 3]
```diff
@@ -155,7 +158,11 @@ protected static ContractTemplate inject(byte [] program, int [] offsets, Parame
 			if (paramIdx < offsets.length && offsets[paramIdx] == progIdx) {
 				ParameterValue value = values[paramIdx];
 				++paramIdx;
-				progIdx += value.updateProgram(updatedProgram);
+				for (byte b : value.toBytes()) {
+					updatedProgram.add(b);
+				}
+				progIdx += value.placeholderSize();
+				//progIdx += value.appendToProgram(updatedProgram);
```
Comment (algonautshant) [inline src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: Is this commented line needed?

Comment (winder) [reply src/main/java/com/algorand/algosdk/templates/ContractTemplate.java]: No, removed.

**Pull request merged**